src/hisp/scenario.py

```diff
@@ -0,0 +1,229 @@
+import pandas as pd
+from typing import List
+import warnings
+
+
+class Pulse:
+    pulse_type: str
+    nb_pulses: int
+    ramp_up: float
+    steady_state: float
+    ramp_down: float
+    waiting: float
+
+    def __init__(
+        self,
+        pulse_type: str,
+        nb_pulses: int,
+        ramp_up: float,
+        steady_state: float,
+        ramp_down: float,
+        waiting: float,
+    ):
+        self.pulse_type = pulse_type
+        self.nb_pulses = nb_pulses
+        self.ramp_up = ramp_up
+        self.steady_state = steady_state
+        self.ramp_down = ramp_down
+        self.waiting = waiting
+
+    @property
+    def total_duration(self) -> float:
+        all_zeros = (
+            self.ramp_up == 0
+            and self.steady_state == 0
+            and self.ramp_down == 0
+            and self.waiting == 0
+        )
+        if self.pulse_type == "RISP" and all_zeros:
+            msg = "RISP pulse has all zeros for ramp_up, steady_state, ramp_down, waiting. "
+            msg += "Setting hardcoded values. Please check the values in the scenario file."
+            warnings.warn(msg, UserWarning)
+
+            self.ramp_up = 10
+            self.steady_state = 250
+            self.ramp_down = 10
+            self.waiting = 1530
+
+        return self.ramp_up + self.steady_state + self.ramp_down + self.waiting
+
+    @property
+    def duration_no_waiting(self) -> float:
+        return self.total_duration - self.waiting
+
+
+class Scenario:
+    def __init__(self, pulses: List[Pulse] = None):
+        """Initializes a Scenario object containing several pulses.
+
+        Args:
+            pulses: The list of pulses in the scenario. Each pulse is a Pulse object.
+        """
+        self._pulses = pulses if pulses is not None else []
+
+    @property
+    def pulses(self) -> List[Pulse]:
+        return self._pulses
+
+    def to_txt_file(self, filename: str):
+        df = pd.DataFrame(
+            [
+                {
+                    "pulse_type": pulse.pulse_type,
+                    "nb_pulses": pulse.nb_pulses,
+                    "ramp_up": pulse.ramp_up,
+                    "steady_state": pulse.steady_state,
+                    "ramp_down": pulse.ramp_down,
+                    "waiting": pulse.waiting,
+                }
+                for pulse in self.pulses
+            ]
+        )
+        df.to_csv(filename, index=False)
+
+    @staticmethod
+    def from_txt_file(filename: str, old_format=False) -> "Scenario":
+        if old_format:
```

Comment:
hopefully this will become obsolete when we expand to new users

Reply:
Yes!